sql/src/main/java/io/crate/execution/dml/upsert/CheckConstraints.java

```diff
@@ -56,6 +67,29 @@
             inputs.add(ctx.add(notNullRef));
         }
         expressions = ctx.expressions();
+        checkConstraints = initializeCheckConstraints(table, inputFactory, ctx);
+    }
+
+    private Map<String, Tuple<Input<?>, String>> initializeCheckConstraints(DocTableInfo table,
+                                                                            InputFactory inputFactory,
+                                                                            InputFactory.Context<E> ctx) {
+        ExpressionAnalyzer analyzer = new ExpressionAnalyzer(
```

Comment:
I think the `DocTableInfo` should expose the checkConstraints in analyzed form (Symbol) - we don't want to have the overhead of analyzing the expressions on each insert.